bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v3/teaser/CommerceTeaserImpl.java

```diff
@@ -0,0 +1,202 @@
+import com.adobe.cq.wcm.core.components.models.ListItem;
+import com.adobe.cq.wcm.core.components.models.Teaser;
+import com.adobe.cq.wcm.core.components.models.datalayer.ComponentData;
+import com.day.cq.wcm.api.Page;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+@Model(
+    adaptables = SlingHttpServletRequest.class,
+    adapters = { Teaser.class, ComponentExporter.class },
+    resourceType = CommerceTeaserImpl.RESOURCE_TYPE)
+@Exporter(
+    name = ExporterConstants.SLING_MODEL_EXPORTER_NAME,
+    extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class CommerceTeaserImpl implements CommerceTeaser {
+    protected static final String RESOURCE_TYPE = "core/cif/components/content/teaser/v3/teaser";
+
+    private List<ListItem> actions = new ArrayList<>();
+
+    @ScriptVariable
+    private Resource resource;
+
+    @ScriptVariable
+    private Page currentPage;
+
+    @OSGiService
+    private UrlProvider urlProvider;
+
+    @Self
+    private SlingHttpServletRequest request;
+
+    @Self
+    @Via(type = ResourceSuperType.class)
+    private Teaser wcmTeaser;
+
+    @PostConstruct
+    void initModel() {
+        if (isActionsEnabled()) {
+            populateActions();
+        }
+    }
+
+    void populateActions() {
+        Resource actionsNode = resource.getChild(CommerceTeaser.NN_ACTIONS);
+        if (actionsNode != null) {
+            Iterable<Resource> configuredActions = actionsNode.getChildren();
+
+            Page productPage = SiteNavigation.getProductPage(currentPage);
+            Page categoryPage = SiteNavigation.getCategoryPage(currentPage);
+
+            // build teaser action items for all configured actions
+            for (Resource action : configuredActions) {
+                ValueMap actionProperties = action.getValueMap();
+                String productSku = actionProperties.get(PN_ACTION_PRODUCT_SKU, String.class);
+                String categoryUid = actionProperties.get(PN_ACTION_CATEGORY_ID, String.class);
+                String link = actionProperties.get(Teaser.PN_ACTION_LINK, String.class);
+
+                String actionUrl = null;
+                CommerceIdentifier identifier = null;
+
+                if (StringUtils.isNotBlank(categoryUid)) {
+                    actionUrl = urlProvider.toCategoryUrl(request, categoryPage, categoryUid);
+                    identifier = new CommerceIdentifierImpl(categoryUid, CommerceIdentifier.IdentifierType.UID,
+                        CommerceIdentifier.EntityType.CATEGORY);
+                } else if (StringUtils.isNotBlank(productSku)) {
+                    actionUrl = urlProvider.toProductUrl(request, productPage, productSku);
+                    identifier = new CommerceIdentifierImpl(productSku, CommerceIdentifier.IdentifierType.SKU,
+                        CommerceIdentifier.EntityType.PRODUCT);
+                } else if (StringUtils.isNotBlank(link)) {
+                    actionUrl = link + ".html";
```

Comment:
Could we use the action from the WCM Teaser instead? This implementation applies some additional logic on the link which is private and we cannot easily replicate it (Calling LinkHandler#getLink())